xl-op/blueprint.yaml

```diff
@@ -8,15 +8,55 @@ metadata:
   version: 1.0
 spec:
   parameters:
+    - name: ImageRegistryType
+      type: Select
+      prompt: "Select the type of Image Registry:"
+      options:
+        - label: Default (Uses various default public image registries for the images)
+          value: default
+        - label: Custom Public Registry (Overrides defaults to use a specific custom public registry)
+          value: public
+        - label: Custom Private Registry - Password protected (Overrides defaults to use a specific custom private registry)
+          value: private
+      promptIf: !expr "ProcessType == 'install' || ProcessType == 'upgrade'"
+      ignoreIfSkipped: true
+      saveInXlvals: true
+      default: DefaultImageRegistry
+      description: Select the type of the Image Registry to use for pulling all images required for the installation.
+    - name: CustomImageRegistryName
+      type: Input
+      saveInXlvals: true
+      ignoreIfSkipped: true
+      overrideDefault: true
+      promptIf: !expr "ImageRegistryType == 'public' || ImageRegistryType == 'private' && (ProcessType == 'install' || ProcessType == 'upgrade')"
```

Comment:
I think this should be:

`promptIf: !expr ("ImageRegistryType == 'public' || ImageRegistryType == 'private') && (ProcessType == 'install' || ProcessType == 'upgrade')"`

?

Reply:
maybe better use IsCustomImageRegistry, instead of ("ImageRegistryType == 'public' || ImageRegistryType == 'private') ?